.github/workflows/check_smoke_tests.yml

```diff
@@ -7,132 +7,182 @@ on:
     branches: [main]
 
 jobs:
+  service-set-up:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        database: [postgres:17, mysql:8, mssql/server:2022-latest]
```

Comment:
One thing we should consider is pinning these databases.  Using "latest" specifically means that we will redownload an image whenever they publish anything new.  Using ":8" or ":17" means we will redownload an image whenever they make a minor release change.  Are those incremental changes necessary for our work?

Reply:
Resolved by adding minor versions!